fix: discover search lag

estebanmino opened the pull request:
Fixes RNBW-3181

## What changed (plus any additional context for devs)

if you search for an ENS name, clear the query and search again multiple instances of the same row repeat
for some reason ExchangeAssetSectionList doesn't update correctly on data change, so I added a key depending on search result to force updates

## PoW (screenshots / screen recordings)

before
https://recordit.co/GsY0MC2n7D

after
https://recordit.co/mJjwwSvL8M


## Dev checklist for QA: what to test

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

skylarbarrera commented on src/components/discover-sheet/DiscoverSearch.js:
```diff
@@ -58,6 +58,14 @@ export default function DiscoverSearch() {
+    () =>
+      uniswapCurrencyList?.[0]?.data?.length ||
+      0 + ensResults?.[0]?.data?.length ||
+      0,
```
nit: can we do a string so theres no possibility of the length being the same? something like `${uniswapCurrencyList?.[0]?.data?.length}_${ensResults?.[0]?.data?.length}`

osdnk replied:
yes, I don't want to be a pain in the ass, but this is not 100% unique

estebanmino replied:
💯  done!

skylarbarrera reviewed (commented):
👌

osdnk commented:
/rebase

estebanmino commented:
@osdnk is that the new rebase gh action?

terrysahaidak commented:
I think the problem is in `CurrencySelectionList`, it has `magicMemo` but not the `query` as its dependency.

terrysahaidak commented:
@estebanmino pinging you regarding my last comment :)

estebanmino commented:
oh sorry! @terrysahaidak   taking a look now

estebanmino commented:
/rebase

estebanmino commented:
@terrysahaidak i was still seeing this happening with the `query` dep in magicMemo feel like something is off but i'm not sure about the internals there, added it anyways. 
I changed the key as @osdnk comment to make it unique on search results taking addresses

estebanmino commented:
waiting for profiles to get merged
